### PR TITLE
Create local_internal_options configuration handler fixture

### LINKED ISSUE #1829: Create common tools to edit local internal options file
### Description

It is necessary to create a custom fixture to edit the `local_internal_option` file. This should be used by every test in order to autoconfiguration.

### Implementation
The desirable fixture needs to write in the `local_internal_options` a custom configuration, defined inside the test and restore the original content of the file.

### PR DESCRIPTION
**Pull request by Rebits**
|Related issue|
|---|
|#1829|

## Description

This PR closes #1829. It adds a common Wazuh `local_internal_options` configuration handler fixture for all integration tests.

## Usage

In order to use this fixture, it is necessary to define a `local_internal_options` variable. This should be a dictionary with the necessary internal options for the test. For example:

```
local_internal_options = {'monitord.rotate_log': '0', 'syscheck.debug': '0' }
```
With this variable defined the fixture will works for each test which calls it.

This fixture is defined for module scope cases. In those tests that changes `local_internal_options` according to a list, it is recommended to use the currently defined fixture.

## Tests

- [x] Proven that this fixture works correctly
- [x] Python codebase satisfies PEP-8 style style guide. `pycodestyle --max-line-length=120 --show-source --show-pep8 file.py`.
- [x] Python codebase is documented following the Google Style for Python docstrings.

### CODE REVIEW

File: tests/integration/conftest.py [hunk 1]
```diff
@@ -602,3 +602,30 @@ def create_file_structure_function(get_files_list):
     yield
 
     delete_file_structure(get_files_list)
+
+
+@pytest.fixture(scope="module")
```
Comment (juliamagan) [inline tests/integration/conftest.py:607]:
```suggestion
@pytest.fixture(scope='module')
```

File: tests/integration/conftest.py [hunk 2]
```diff
@@ -602,3 +602,30 @@ def create_file_structure_function(get_files_list):
...
+    try:
+        local_internal_options = getattr(request.module, 'local_internal_options')
+    except AttributeError as local_internal_configuration_not_set:
+        logger.error('Configure Local Internal Options: Error - local_internal_options is not set')
+        raise local_internal_configuration_not_set
+
+    backup_local_internal_options = conf.get_local_internal_options_dict()
+
+    logger.error('Configure Local Internal Options: DEBUG - Set local_internal_option to '  +
+                    f"{str(local_internal_options)}")
```
Comment (juliamagan) [inline tests/integration/conftest.py:624]: This logger lever should be `info` or `debug`

File: tests/integration/conftest.py [hunk 3]
```diff
@@ -602,3 +602,30 @@ def create_file_structure_function(get_files_list):
...
+    conf.set_local_internal_options_dict(local_internal_options)
+
+    yield
+
+    logger.error('Configure Local Internal Options: DEBUG - Restore local_internal_option to '  +
+                    f"{str(backup_local_internal_options)}")
```
Comment (juliamagan) [inline tests/integration/conftest.py:630]: Same as before

**Review by juliamagan — CHANGES_REQUESTED** [review]
GJ, some changes are required

**Comment by juliamagan** [comment]
We can test these changes before the merge in temporal branch: `common-fixtures-testing`

File: tests/integration/conftest.py [hunk 4]
```diff
@@ -602,3 +602,30 @@ def create_file_structure_function(get_files_list):
...
+    """
+    try:
+        local_internal_options = getattr(request.module, 'local_internal_options')
+    except AttributeError as local_internal_configuration_not_set:
+        logger.error('Configure Local Internal Options: Error - local_internal_options is not set')
+        raise local_internal_configuration_not_set
+
+    backup_local_internal_options = conf.get_local_internal_options_dict()
+
+    logger.error('Configure Local Internal Options: DEBUG - Set local_internal_option to ' +
```
Comment (jmv74211) [inline tests/integration/conftest.py]: Change this to `logger.debug`

Comment (jmv74211) [reply tests/integration/conftest.py]: `+` is not neccessary

File: tests/integration/conftest.py [hunk 5]
```diff
@@ -602,3 +602,30 @@ def create_file_structure_function(get_files_list):
...
+                 f"{str(local_internal_options)}")
+    conf.set_local_internal_options_dict(local_internal_options)
+
+    yield
+
+    logger.error('Configure Local Internal Options: DEBUG - Restore local_internal_option to ' +
```
Comment (jmv74211) [inline tests/integration/conftest.py]: Change this to `logger.debug`

File: tests/integration/conftest.py (hunk 3 above)
Comment (Rebits) [reply tests/integration/conftest.py]: Done 87039c12c10a6430991b78dae1796a53496c1600

File: tests/integration/conftest.py (hunk 2 above)
Comment (Rebits) [reply tests/integration/conftest.py]: Done 87039c12c10a6430991b78dae1796a53496c1600

File: tests/integration/conftest.py (hunk 1 above)
Comment (Rebits) [reply tests/integration/conftest.py]: Done 87039c12c10a6430991b78dae1796a53496c1600

**Review by jmv74211 — CHANGES_REQUESTED** [review]
GJ some requested changes. I think you would have to make a custom class for the logger and use an appropriate format.

File: deps/wazuh_testing/wazuh_testing/tools/__init__.py [hunk 6]
```diff
@@ -146,3 +146,21 @@ def get_service():
...
+                                        'agent.debug': '2'}.update(DISABLE_MONITORD_ROTATE_LOG_OPTION)
+FIM_LOCAL_INTERNAL_OPTIONS_AGENT_WINDOWS = {'syscheck.debug': '2',
+                                            'windows.debug': '2'}.update(DISABLE_MONITORD_ROTATE_LOG_OPTION)
+GCLOUD_LOCAL_INTERNAL_OPTIONS = {'analysisd.debug': '2',
+                                'wazuh_modules.debug': '2'}.update(DISABLE_MONITORD_ROTATE_LOG_OPTION)
+LOGTEST_LOCAL_INTERNAL_OPTIONS = {'analysisd.debug': '2'}
+REMOTED_LOCAL_INTERNAL_OPTIONS = {'remoted.debug': '2', 'wazuh_database.interval': '2', 'wazuh_db.commit_time': '2',
+                                    'wazuh_db.commit_time_max': '3' }.update(DISABLE_MONITORD_ROTATE_LOG_OPTION)
+VD_LOCAL_INTERNAL_OPTIONS = {'wazuh_modules.debug': '2'}.update(DISABLE_MONITORD_ROTATE_LOG_OPTION)
+WPK_LOCAL_INTERNAL_OPTIONS = {'wazuh_modules.debug': '2'}
```
[the remark this answers is not in the document]
Comment (Rebits) [reply deps/wazuh_testing/wazuh_testing/tools/__init__.py:166]: Done 
453f1c6

File: tests/integration/conftest.py (hunk 5 above)
Comment (Rebits) [reply tests/integration/conftest.py]: Done 87039c1

**Review by jmv74211 — APPROVED** [review]

**Review by juliamagan — APPROVED** [review]

**Pull request merged**